src/approaches/interactive_learning_approach.py

```diff
@@ -88,28 +90,27 @@ def learn_from_offline_dataset(self, dataset: Dataset,
                 self._simulator,
                 self._get_current_predicates(),
                 max_steps=CFG.interactive_max_steps)
-            # Pick best state seen during exploration
-            abstract_states = [
-                utils.abstract(s, self._get_current_predicates())
-                for s in traj.states
-            ]
-            best_abstract_state = get_best_goal_by_score(
-                self._dataset_with_atoms, abstract_states)
-            idx = abstract_states.index(best_abstract_state)
-            best_state = traj.states[idx]
-            # Pick best ground atom in the state
-            ground_atoms = utils.all_possible_ground_atoms(
-                best_state, self._predicates_to_learn)
-            best_atom_set = get_best_goal_by_score(self._dataset_with_atoms,
-                                                   [{ga}
-                                                    for ga in ground_atoms])
-            assert len(best_atom_set) == 1
-            # For now, ask teacher about every such best atom
-            if self._ask_teacher(best_state, list(best_atom_set)[0]):
-                # Add this atom if it's a positive example
-                self._dataset_with_atoms.append(  # pragma: no cover
-                    (LowLevelTrajectory([best_state], []), [best_atom_set]))
-                # Still need to implement a way to use negative examples
+            # Decide whether to ask at each state during exploration
+            for s in traj.states:
+                abstract_state = utils.abstract(s,
+                                                self._get_current_predicates())
+                score = score_goal(self._dataset_with_atoms, abstract_state)
```

Comment:
so `score_goal` was originally intended for use with goals sampled from glib, right? but now we're passing in the whole `abstract_state` instead of just a goal.

furthermore, based on our slack conversation just now, i think we should refactor score functions so that they take in single atoms, rather than multiple. in doing so, we can change this function to `score_atom`. then we don't need this two-step process where we first decide whether to ask about a whole state, and then decide about each atom -- instead, we just make one judgement about each atom.